cmake/modules/AddSwift.cmake

```diff
@@ -1568,440 +1561,409 @@ function(add_swift_library name)
-              add_dependencies("swift-test-stdlib${VARIANT_SUFFIX}"
-                  ${lipo_target}
-                  ${lipo_target_static})
-            endif()
           endif()
-        endforeach()
-      endif()
-    endforeach()
-  else()
-    set(sdk "${SWIFT_HOST_VARIANT_SDK}")
-    set(arch "${SWIFT_HOST_VARIANT_ARCH}")
-
-    # Collect compiler flags
-    set(swiftlib_swift_compile_flags_all ${SWIFTLIB_SWIFT_COMPILE_FLAGS})
-    if("${sdk}" STREQUAL "OSX")
-      list(APPEND swiftlib_swift_compile_flags_all
-        ${SWIFTLIB_SWIFT_COMPILE_FLAGS_OSX})
-    elseif("${sdk}" STREQUAL "IOS" OR "${sdk}" STREQUAL "IOS_SIMULATOR")
-      list(APPEND swiftlib_swift_compile_flags_all
-        ${SWIFTLIB_SWIFT_COMPILE_FLAGS_IOS})
-    elseif("${sdk}" STREQUAL "TVOS" OR "${sdk}" STREQUAL "TVOS_SIMULATOR")
-      list(APPEND swiftlib_swift_compile_flags_all
-        ${SWIFTLIB_SWIFT_COMPILE_FLAGS_TVOS})
-    elseif("${sdk}" STREQUAL "WATCHOS" OR "${sdk}" STREQUAL "WATCHOS_SIMULATOR")
-      list(APPEND swiftlib_swift_compile_flags_all
-        ${SWIFTLIB_SWIFT_COMPILE_FLAGS_WATCHOS})
+        endif()
+      endforeach()
     endif()
+  endforeach()
+endfunction()
 
-    _add_swift_library_single(
-      ${name}
-      ${name}
-      ${SWIFTLIB_SHARED_keyword}
-      ${SWIFTLIB_STATIC_keyword}
-      ${SWIFTLIB_OBJECT_LIBRARY_keyword}
-      ${SWIFTLIB_SOURCES}
-      SDK ${sdk}
-      ARCHITECTURE ${arch}
-      DEPENDS ${SWIFTLIB_DEPENDS}
-      LINK_LIBRARIES ${SWIFTLIB_LINK_LIBRARIES}
-      FRAMEWORK_DEPENDS ${SWIFTLIB_FRAMEWORK_DEPENDS}
-      FRAMEWORK_DEPENDS_WEAK ${SWIFTLIB_FRAMEWORK_DEPENDS_WEAK}
-      LLVM_COMPONENT_DEPENDS ${SWIFTLIB_LLVM_COMPONENT_DEPENDS}
-      FILE_DEPENDS ${SWIFTLIB_FILE_DEPENDS}
-      C_COMPILE_FLAGS ${SWIFTLIB_C_COMPILE_FLAGS}
-      SWIFT_COMPILE_FLAGS ${swiftlib_swift_compile_flags_all}
-      LINK_FLAGS ${SWIFTLIB_LINK_FLAGS}
-      PRIVATE_LINK_LIBRARIES ${SWIFTLIB_PRIVATE_LINK_LIBRARIES}
-      INTERFACE_LINK_LIBRARIES ${SWIFTLIB_INTERFACE_LINK_LIBRARIES}
-      INCORPORATE_OBJECT_LIBRARIES ${SWIFTLIB_INCORPORATE_OBJECT_LIBRARIES}
-      INCORPORATE_OBJECT_LIBRARIES_SHARED_ONLY ${SWIFTLIB_INCORPORATE_OBJECT_LIBRARIES_SHARED_ONLY}
-      ${SWIFTLIB_DONT_EMBED_BITCODE_keyword}
-      ${SWIFTLIB_API_NOTES_NON_OVERLAY_keyword}
-      ${SWIFTLIB_IS_STDLIB_keyword}
-      ${SWIFTLIB_IS_STDLIB_CORE_keyword}
-      ${SWIFTLIB_IS_SDK_OVERLAY_keyword}
-      INSTALL_IN_COMPONENT "${SWIFTLIB_INSTALL_IN_COMPONENT}"
-      DEPLOYMENT_VERSION_OSX "${SWIFTLIB_DEPLOYMENT_VERSION_OSX}"
-      DEPLOYMENT_VERSION_IOS "${SWIFTLIB_DEPLOYMENT_VERSION_IOS}"
-      DEPLOYMENT_VERSION_TVOS "${SWIFTLIB_DEPLOYMENT_VERSION_TVOS}"
-      DEPLOYMENT_VERSION_WATCHOS "${SWIFTLIB_DEPLOYMENT_VERSION_WATCHOS}"
-      )
+function(add_swift_library name)
+  set(ASL_options FORCE_BUILD_OPTIMIZED)
+  set(ASL_single_parameter_options)
+  set(ASL_multiple_parameter_options GYB_SOURCES)
```

Comment:
I love the `GYB_SOURCES` argument! Great addition here to separate out the sources.